### DISCUSSION TITLE
Proposal: Operators as method groups.

### DISCUSSION THREAD

**Discussion by orthoxerox** [dotnet/csharplang#598] [Language Ideas]
It would be quite convenient if standalone operators were treated as method groups. Instead of `xs.Aggregate((x1, x2) => x1 + x2)` you would write `xs.Aggregate(+)`. If that's potentially syntactically ambiguous, the operator symbol could be wrapped in parentheses or prefixed with `@` or another symbol.

**Comment by CyrusNajmabadi** [comment]
I think being able to say "Type.new" to be able to reference a constructor as a method group would also be nice.

**Comment by YaakovDavis** [comment]
@CyrusNajmabadi 
What about `Type.ctor`? It's more descriptive IMO.

What about argumented ctors? Possible syntax:
`Foo.ctor<int, string>`

**Comment by CyrusNajmabadi** [comment]
Type.ctor is already a legal expression.  Type.new is not.  :)

> What about argumented ctors

We don't have any special syntax to pick a particular overload of a method group.  So i don't htink we'd have anything for constructors.

**Comment by alrz** [comment]
Both `new` and operators are also mentioned in https://github.com/dotnet/csharplang/issues/84.

**Comment by Pzixel** [comment]
Is it possible to make operators to being working with null propagation (as method calls, for example), e..g:

        private int Test(int? value) => value?.+(5) ?? 0;

which is the equivalent of:

        private int Test(int? value) => value?.Add(5) ?? 0;

**Comment by Joe4evr** [comment]
Operators already work on Nullable value types, last I read.

**Comment by Pzixel** [comment]
You're right. Then it's just 

```csharp
private int Test(int? value) => (value + 5) ?? 0;
```

However, it's a compiler magic. It wont work for the custom code


```csharp
private int Test(MyClass value) => (value + 5) ?? 0;

class MyClass
{
	public int Value { get; }
	
	public MyClass(int value) => Value = value;
	
	public static MyClass operator +(MyClass a, int value) => new MyClass(a.Value + value);
}
```
